File: python/burkut/resources/funds.py
"""
Investment Funds (TEFAS) Resource
"""

from typing import Any, Dict, List
from burkut.resources.base import BaseResource


class FundsResource(BaseResource):
    """Yatırım Fonları (TEFAS) verileri."""

    def list(self) -> List[Dict[str, Any]]:
        """
        Tüm yatırım fonlarını listeler.

        Returns:
            Yatırım fonları listesi.
        """
        res = self._get("/funds")
        return res if isinstance(res, list) else []

    def get(self, symbol: str) -> Dict[str, Any]:
        """
        Belirli bir yatırım fonunun detaylarını getirir.

        Args:
            symbol: Fon kodu (Örn: 'TCD', 'AFT', 'MAC')
        """
        if not symbol or not isinstance(symbol, str):
            raise ValueError("symbol parametresi boş olamaz.")
        return self._get(f"/funds/{symbol.strip().upper()}")

File: nodejs/src/resources/funds.ts
import { BaseResource } from './base';
import type { FundItem } from '../types';

export class FundsResource extends BaseResource {
  /**
   * Tüm TEFAS yatırım fonlarını listeler.
   */
  async list(): Promise<FundItem[]> {
    return this._get<FundItem[]>('/funds');
  }

  /**
   * Belirli bir TEFAS yatırım fonunun detaylarını getirir.
   * @param symbol Fon kodu (Örn: 'TCD', 'AFT', 'MAC')
   */
  async get(symbol: string): Promise<FundItem> {
    if (!symbol || typeof symbol !== 'string') {
      throw new Error('symbol parametresi geçerli bir metin olmalıdır.');
    }
    return this._get<FundItem>(`/funds/${symbol.trim().toUpperCase()}`);
  }
}
